aggregator.py

```diff
@@ -498,6 +498,14 @@ def parse_event_packet(self, packet):
             raise Exception(u'Unparseable event packet: %s' % packet)
 
     def submit_packets(self, packets):
+        # Usually we should always decode the string
+        # as utf-8 because `packets` passed through a
+        # network socket, but if submit_packets is used
+        # programatically and packets is unicode already
+        # then do not decode!
+        if not isinstance(packets, unicode):
```

Comment:
Could you use:
```python
if not type(packets) == unicode:
```
instead ?

It's slightly faster:
```python
topdog@i-1b871af7(prod):~$ python -m timeit 'not isinstance("sdfsdfsd" , unicode)'
10000000 loops, best of 3: 0.197 usec per loop
```

```python
topdog@i-1b871af7(prod):~$ python -m timeit 'not type("sdfsdfsd") == unicode'
10000000 loops, best of 3: 0.151 usec per loop
```

and this code is run in a pretty performance sensitive loop.